src/status_im2/contexts/chat/bottom_sheet_composer/actions/view.cljs

```diff
@@ -0,0 +1,141 @@
+           z-index     (reagent/atom 0)]
+       [:f>
+        (fn []
+          (rn/use-effect (fn []
+                           (if (or (not-empty @text-value) images?)
+                             (when-not (= @z-index 1)
+                               (reset! z-index 1)
+                               (js/setTimeout #(reanimated/animate btn-opacity 1) 50))
+                             (when-not (= @z-index 0)
+                               (reanimated/animate btn-opacity 0)
+                               (js/setTimeout #(reset! z-index 0) 300))))
+                         [@text-value])
+          [reanimated/view
+           {:style (style/send-button btn-opacity @z-index)}
+           [quo/button
+            {:icon                true
+             :size                32
+             :accessibility-label :send-message-button
+             :on-press            #(send-message props state animations window-height)}
+            :i/arrow-up]])]))])
+
+(defn audio-button
+  []
+  [quo/button
+   {:on-press #(js/alert "to be added")
+    :icon     true
+    :type     :outline
+    :size     32}
+   :i/audio])
+
+(defn camera-button
+  []
+  [quo/button
+   {:on-press #(js/alert "to be implemented")
+    :icon     true
+    :type     :outline
+    :size     32
+    :style    {:margin-right 12}}
+   :i/camera])
+
+(defn image-button
+  [{:keys [input-ref]}
+   {:keys [focused?]}
+   {:keys [height]}
+   insets]
+  [quo/button
+   {:on-press (fn []
+                (permissions/request-permissions
+                 {:permissions [:read-external-storage :write-external-storage]
+                  :on-allowed  (fn []
+                                 (when platform/android?
+                                   (when @focused?
+                                     (rf/dispatch [:chat.ui/set-input-refocus true]))
+                                   (.blur ^js @input-ref))
+                                 (rf/dispatch [:chat.ui/set-input-content-height
+                                               (reanimated/get-shared-value height)])
+                                 (rf/dispatch [:open-modal :photo-selector {:insets insets}]))
+                  :on-denied   (fn []
+                                 (alert/show-popup (i18n/label :t/error)
+                                                   (i18n/label
+                                                    :t/external-storage-denied)))}))
+    :icon     true
+    :type     :outline
+    :size     32
+    :style    {:margin-right 12}}
+   :i/image])
+
+(defn reaction-button
+  []
+  [quo/button
+   {:on-press #(js/alert "to be implemented")
+    :icon     true
+    :type     :outline
+    :size     32
+    :style    {:margin-right 12}}
+   :i/reaction])
+
+(defn format-button
+  []
+  [quo/button
+   {:on-press #(js/alert "to be implemented")
+    :icon     true
+    :type     :outline
+    :size     32}
+   :i/format])
+
+(defn view
+  [props state animations window-height insets images?]
```

Comment:
`props`, `state` and `animations` are already maps (and big maps). I think maps make since when building a reusable component that be used in multiple places -- it makes it clear what keys are being used. But here this component is meant to be used only once in the composer, so using a map would just lengthen the call of the component IMO.

Reply:
I agree, , it's worst to merge everything inside a single map just to call a component.

Reply:
it should be one map